industry_restaurant/data/pos_payment_method.xml

```diff
@@ -0,0 +1,16 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo noupdate="1">
+    <record id="pos_payment_method_1" model="pos.payment.method">
+        <field name="name">Sodexo Card Payment</field>
+        <field name="journal_id" ref="account.1_bank"></field>
+    </record>
+    <record id="pos_payment_method_2" model="pos.payment.method">
+        <field name="name">Online Payment</field>
+        <field name="is_online_payment" eval="1"/>
+        <field name="online_payment_provider_ids" eval="[
+            (6, 0, [
+                ref('payment.payment_provider_demo'),
+            ])
+        ]"/>
+    </record>
```

Comment:
We shouldn't play with demo provider outside of demo data

Reply:
So for this payment provider is itself in data file in base code Ref link - https://github.com/odoo/odoo/blob/17.0/addons/payment/data/payment_provider_data.xml#L209-L214 
and I have also checked by creating a new db without demo data and its working fine over there also

Reply:
Yes, but here you're linking the method to the pos config in data... I recommend to link it in demo only